tree/treeplayer/src/TTreeFormula.cxx

```diff
@@ -382,7 +386,7 @@ void TTreeFormula::DefineDimensions(Int_t code, Int_t size,
 
    fCumulSizes[code][fNdimensions[code]] = size;
 
-   if ( fIndexes[code][fNdimensions[code]] < 0 ) {
+   if ( !scalarindex && fIndexes[code][fNdimensions[code]] < 0 ) {
```

Comment:
It's hard to say from reading the code, but it looks like `scalarindex` is equivalent to `vsize == 1`. Maybe it's unnecessary.

Reply:
vsize==1 could also be a std::vector that happen to have one element in this entry.